### ISSUE TITLE
Roslyn doesn't understand 'string.IsNullOrWhiteSpace(x) == false' null check

### ISSUE THREAD

**Issue by chris5287** [dotnet/roslyn#39298]
**Version Used**: 
.NET Core SDK 3.0.100

**Steps to Reproduce**:
Compile the following code:

```cs
#nullable enable
class C
{
    string? field;

    void M1(C c)
    {
        if (!string.IsNullOrWhiteSpace(c.field))
        {
            c.field.ToString();
        }
    }

    void M2(C c)
    {
        if (string.IsNullOrWhiteSpace(c.field) == false)
        {
            c.field.ToString(); // incorrect CS8602
        }
    }
}
```

**Expected Behavior**:
Both methods report no warnings

**Actual Behavior**:
`M2` has an incorrect warning `CS8602: Dereference of a possibly null reference` reported for `c.field.ToString()`

Might be related to #39264 ?

**Comment by jcouv** [comment]
It looks like the `IsNullOrWhiteSpace` method is correctly annotated.
The issue is the `== false` throws off the analysis (known issue, https://github.com/dotnet/roslyn/issues/29855). Use `if (!string.IsNullOrWhiteSpace(c.field)) ... ` instead.

Closing as duplicate of https://github.com/dotnet/roslyn/issues/29855
Thanks
